sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -624,6 +624,9 @@ class Analyzer(
       } catch {
         case _: NoSuchTableException =>
           u.failAnalysis(s"Table or view not found: ${u.tableName}")
+        // If the database is not valid, also throw an AnalysisException.
+        case _: NoSuchDatabaseException =>
+          u.failAnalysis(s"Table or view not found: ${u.tableName}")
```

Comment:
shall we use the qualified name? or users are hard to find out that the database doesn't exist, given only the table name.